src/main/java/io/cdap/plugin/successfactors/source/transport/SuccessFactorsTransporter.java

```diff
@@ -189,6 +197,39 @@ private Request buildRequest(URL endpoint, String mediaType) {
       .build();
   }
 
+  /**
+   * Builds and configures an OkHttpClient with the specified proxy settings and authentication credentials.
+   * @param proxyUrl The URL of the proxy server (e.g., "http://proxy.example.com:8080").
+   * Set to null or an empty string to bypass proxy configuration.
+   * @param proxyUsername  The username for proxy authentication. Set to null or an empty string if not required.
+   * @param proxyPassword  The password for proxy authentication. Set to null or an empty string if not required.
+   * @return An OkHttpClient configured with the specified proxy settings and authentication credentials.
+   */
+  private OkHttpClient buildConfiguredClient(String proxyUrl, String proxyUsername, String proxyPassword)
+    throws MalformedURLException, TransportException {
+    OkHttpClient.Builder builder = getConfiguredClient();
+
+    if (proxyUrl != null && !proxyUrl.isEmpty()) {
+      URL url = new URL(proxyUrl);
+      Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(url.getHost(), url.getPort()));
+      builder.proxy(proxy);
+
+      if (SuccessFactorsUtil.isNotNullOrEmpty(proxyUsername) && SuccessFactorsUtil.isNotNullOrEmpty(proxyPassword)) {
```

Comment:
use 
`!Strings.isNullOrEmpty(proxyUrl)`

Reply:
SuccessFactorsUtil.isNotNullOrEmpty, this is being used in the whole project